gammapy/spectrum/results.py

```diff
@@ -146,6 +148,37 @@ def __init__(self, spectral_model, parameters, parameter_errors,
         self.parameters.reference = self.parameters.reference.to('TeV')
         self.parameter_errors.reference = self.parameter_errors.reference.to('TeV')
 
+    @classmethod
+    def from_3fgl(cls, source):
+        """Retrieve spectral fit result from 3FGL source"""
+        d = source.data
+        parameters = dict()
+        parameter_errors = dict()
+        spectral_model = d['SpectrumType'].strip()
+        if spectral_model == 'PowerLaw':
+            parameters['index'] = Quantity(d['Spectral_Index'], '')
+            parameter_errors['index'] = Quantity(d['Unc_Spectral_Index'], '')
+            parameters['reference'] = Quantity(d['Pivot_Energy'], 'MeV')
+            parameter_errors['reference'] = Quantity(0, 'MeV')
+            parameters['norm'] = Quantity(d['Flux_Density'], 'cm-2 s-1 MeV-1')
+            parameter_errors['norm'] = Quantity(d['Unc_Flux_Density'],
+                                                'cm-2 s-1 MeV-1')
+        elif spectral_model == 'LogParabola':
+            parameters['alpha'] = Quantity(d['Spectral_Index'], '')
+            parameter_errors['alpha'] = Quantity(d['Unc_Spectral_Index'], '')
+            parameters['beta'] = Quantity(d['beta'], '')
+            parameter_errors['beta'] = Quantity(d['Unc_beta'], '')
+            parameters['reference'] = Quantity(d['Pivot_Energy'], 'MeV')
+            parameter_errors['reference'] = Quantity(0, 'MeV')
+            parameters['norm'] = Quantity(d['Flux_Density'], 'cm-2 s-1 MeV-1')
+            parameter_errors['norm'] = Quantity(d['Unc_Flux_Density'],
+                                                'cm-2 s-1 MeV-1')
+        else:
```

Comment:
I meant call unique on the Fermi catalog flux model column to get a complete list of all models there are, so that it's clear what further work is needed.
Not call unique here in the code.